### ISSUE TITLE
P3D on Linux issue with Intel integrated graphics due to out of date JOGL: Profile GL4bc is not available on X11GraphicsDevice

### ISSUE THREAD

**Issue by Kyle-Gagner** [processing/processing#6160]
## Description

Attempting to use P3D causes a runtime exception unless software rendering is enabled. This issue depends on the environment (probably only affects Intel integrated graphics on Linux).

## Expected Behavior

P3D should work.

## Current Behavior

```
java.lang.RuntimeException: Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7f7130021e80, owner true, ResourceToolkitLock[obj 0x6d36e00f, isOwner true, <283473cd, 3bb280c7>[count 1, qsz 0, owner <main-FPSAWTAnimator#00-Timer0>]]], but: [GLProfile[GLES1/GLES1.hw], GLProfile[GLES2/GLES3.hw], GLProfile[GL2ES1/GLES1.hw], GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GLES3/GLES3.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw], GLProfile[GL2GL3/GL4.hw]]
	at processing.opengl.PSurfaceJOGL$2.run(PSurfaceJOGL.java:410)
	at java.lang.Thread.run(Thread.java:748)
RuntimeException: Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7f7130021e80, owner true, ResourceToolkitLock[obj 0x6d36e00f, isOwner true, <283473cd, 3bb280c7>[count 1, qsz 0, owner <main-FPSAWTAnimator#00-Timer0>]]], but: [GLProfile[GLES1/GLES1.hw], GLProfile[GLES2/GLES3.hw], GLProfile[GL2ES1/GLES1.hw], GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GLES3/GLES3.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw], GLProfile[GL2GL3/GL4.hw]]
```

## Steps to Reproduce

```
void setup() {
  size(640, 480, P3D);
}
```
1. launch processing normally `/<path>/processing`, run the above program
2. launch processing with `LIBGL_ALWAYS_SOFTWARE=true /<path>/processing`, run the program

## Your Environment

 

* Processing version: 3.5.4
* Operating System and OS version: Ubuntu 20.04
* Other information: Dell XPS 13 2020 Developer Edition (Intel Iris Plus graphics)

## Possible Causes / Solutions

Most likely, JOGL needs to be updated from 2.3 to 2.4, according to this thread: http://forum.jogamp.org/jogl-problems-on-mesa-20-branch-td4040546.html

**Comment by babaMar** [comment]
Same here

**Comment by benfry** [comment]
Have you tried using the 4.x alpha releases?

**Comment by benfry** [comment]
Should be fixed in the 4.0 releases: https://github.com/processing/processing4/releases
